core/src/main/java/io/fabric8/maven/core/util/kubernetes/KubernetesResourceUtil.java

```diff
@@ -701,6 +701,7 @@ public static void mergePodSpec(PodSpecBuilder builder, PodSpec defaultPodSpec,
             }
             builder.withContainers(containers);
         }
+        return defaultApplicationContainerName; // Return the main application container's name.
```

Comment:
This can return a `null`, is it fine?

Reply:
I am checking null return value in ControllerViaPluginConfigurationEnricher, null value would indicate that enricher hasn't processed(merging default generated container spec with spec provided in resource fragments) controller. It's just here to make us aware of which container is main application container.

If container is specified in resource fragment, this would be set. Otherwise it would be null, in that case FABRIC8_GENERATED processing instruction would be set via handlers